main.go

```diff
@@ -101,12 +101,15 @@ func run(ctx context.Context, wg *sync.WaitGroup, globalCollectionOpts state.Col
 		state.WriteStateFile(servers, globalCollectionOpts, logger)
 	}
 
+	if globalCollectionOpts.TestRun {
+		logger.PrintInfo("Running collector test with %s", util.CollectorNameAndVersion)
+	}
+
 	checkAllInitialCollectionStatus(servers, globalCollectionOpts, logger)
 
 	// We intentionally don't do a test-run in the normal mode, since we're fine with
 	// a later SIGHUP that fixes the config (or a temporarily unreachable server at start)
 	if globalCollectionOpts.TestRun {
-		logger.PrintInfo("Running collector test with %s", util.CollectorNameAndVersion)
```

Comment:
I moved this because otherwise the warning is printed before the test run output, which seems like it should be the first thing. I don't love this, but I couldn't think of a better way without significant refactoring. Ideas welcome.

Reply:
I think this is fine (we may be able to improve wording or styling a bit), but I'm now curious what you meant by "I don't love this, but I couldn't think of a better way without significant refactoring", like what kind of refactoring you are thinking

Reply:
I wasn't thinking of anything specific, but I guess maybe extracting the test code to a function? Though thinking about this, we'd still need to special-case the initial test output, so maybe this is moot. I'll just go with this for now and we can revisit.